pkg/controllers/infracluster/azure.go

```diff
@@ -0,0 +1,205 @@
+	}
+
+	if r.Infra.Status.PlatformStatus == nil {
+		return nil, fmt.Errorf("infrastructure PlatformStatus should not be nil: %w", err)
+	}
+
+	azureClientID, ok := capzManagerBootstrapCredentials.Data["azure_client_id"]
+	if !ok {
+		return nil, fmt.Errorf("failed to get azureClientID: %w", err)
+	}
+
+	azureTenantID, ok := capzManagerBootstrapCredentials.Data["azure_tenant_id"]
+	if !ok {
+		return nil, fmt.Errorf("failed to get azureTenantID: %w", err)
+	}
+
+	providerSpec, err := getAzureMAPIProviderSpec(ctx, r.Client)
+	if err != nil {
+		return nil, fmt.Errorf("error obtaining Azure Provider Spec: %w", err)
+	}
+
+	location, err := r.getAzureLocation(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error obtaining Azure Cluster location: %w", err)
+	}
+
+	azureClusterIdentity := &azurev1.AzureClusterIdentity{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.Infra.Status.InfrastructureName,
+			Namespace: defaultCAPINamespace,
+			Annotations: map[string]string{
+				// The ManagedBy Annotation is set so CAPI infra providers ignore the InfraCluster object,
+				// as that's managed externally, in this case by the cluster-capi-operator's infracluster controller.
+				clusterv1.ManagedByAnnotation: managedByAnnotationValueClusterCAPIOperatorInfraClusterController,
+			},
+		},
+		Spec: azurev1.AzureClusterIdentitySpec{
+			Type:              azurev1.ServicePrincipal,
+			AllowedNamespaces: &azurev1.AllowedNamespaces{NamespaceList: []string{defaultCAPINamespace}},
+			ClientID:          string(azureClientID),
+			TenantID:          string(azureTenantID),
+			ClientSecret:      corev1.SecretReference{Name: clusterSecretName, Namespace: defaultCAPINamespace},
+		},
+	}
+
+	if err := r.Create(ctx, azureClusterIdentity); err != nil && !apierrors.IsAlreadyExists(err) {
+		return nil, fmt.Errorf("failed to create Azure Cluster Identity: %w", err)
+	}
+
+	target = &azurev1.AzureCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.Infra.Status.InfrastructureName,
+			Namespace: defaultCAPINamespace,
+			// The ManagedBy Annotation is set so CAPI infra providers ignore the InfraCluster object,
+			// as that's managed externally, in this case by this controller.
+			Annotations: map[string]string{
+				clusterv1.ManagedByAnnotation: managedByAnnotationValueClusterCAPIOperatorInfraClusterController,
+			},
+		},
+
+		Spec: azurev1.AzureClusterSpec{
+			AzureClusterClassSpec: azurev1.AzureClusterClassSpec{
+				Location:         location,
+				AzureEnvironment: "AzurePublicCloud",
+				IdentityRef: &corev1.ObjectReference{
+					Name:      r.Infra.Status.InfrastructureName,
+					Namespace: defaultCAPINamespace,
+					Kind:      "AzureClusterIdentity",
+				},
+			},
+			NetworkSpec: azurev1.NetworkSpec{
+				NodeOutboundLB: &azurev1.LoadBalancerSpec{
+					Name: r.Infra.Status.InfrastructureName,
+					BackendPool: azurev1.BackendPool{
+						Name: r.Infra.Status.InfrastructureName,
+					},
```

Comment:
Is this always true? How did we work this out?

Reply:
Looking into this!

Reply:
Did we get anywhere with this?